config/routes.rb

```diff
@@ -2,5 +2,7 @@
   devise_for :users
   root to: 'pages#home'
 
-  resources :users, only: %i[show edit update]
+  resources :users, only: %i[show edit update] do
+      resources :groups, only: [:index]
```

Comment:
if you do like this it means the url will be /users/:user_id/groups, meaning we'll get a list of all groups _belonging to a user_
but that's not what we want, what we want is a list of all groups, period (plus the filtering mentioned above)